.github/workflows/sub-ci-unit-tests-docker.yml

```diff
@@ -151,19 +151,6 @@ jobs:
       test_variables: '-e NETWORK'
       network: 'Mainnet'
 
-  # Test reconfiguring the the docker image for tesnet.
-  test-configuration-file-testnet:
-    name: Test CI testnet Docker config file
-    # Make sure Zebra can sync the genesis block on testnet
-    uses: ./.github/workflows/sub-test-zebra-config.yml
-    with:
-      test_id: 'testnet-conf'
-      docker_image: ${{ vars.GAR_BASE }}/${{ vars.CI_IMAGE_NAME }}@${{ inputs.image_digest }}
-      grep_patterns: '-e "net.*=.*Test.*estimated progress to chain tip.*Genesis" -e "net.*=.*Test.*estimated progress to chain tip.*BeforeOverwinter"'
-      # TODO: improve the entrypoint to avoid using `ENTRYPOINT_FEATURES=""`
-      test_variables: '-e NETWORK -e ZEBRA_CONF_PATH="/etc/zebrad/zebrad.toml" -e ENTRYPOINT_FEATURES=""'
-      network: 'Testnet'
-
```

Comment:
This test is failing with this PR, similarly to the other one. I have a better approach in mind, which I didn't do in this PR. Let's add it back in a subsequent PR?

Moreover, there seem to be some parts that are bugs or hard to understand. For example, I couldn't figure out what `-e NETWORK` in `test_variables` is supposed to do. Also, setting `ENTRYPOINT_FEATURES" to the empty string to enable the test in the entrypoint makes it very hard to follow the execution path in the whole pipeline.

Reply:
`-e NETWORK` tells docker to use whatever value the `$NETWORK` env variable is set to, to override any default that was set at build time. This happens here: https://github.com/ZcashFoundation/zebra/blob/46c6b6eb38617c07e8a130650b33831759b73263/.github/workflows/sub-test-zebra-config.yml#L90-L91 and makes Zebra run with a Testnet configuration, and the test validates that Zebra is correctly running with it.

> This has saved me (and others) from making mistakes multiple times while making CI refactors, so this is very important under those circumstances.

I do agree that setting the `ENTRYPOINT_FEATURES` to an empty string is a dirty hack to make this work, but that's a tech debt that wouldn't justify removing the whole test. In any case, we can remove the use of the `ENTRYPOINT_FEATURES` variables, while keeping this test behavior.

I'd suggest commenting this and adding a TODO in top of it, or creating an issue, instead of removing the test. Just so we don't forget later on, as this is an important validation.